app/tests/src/wallet/account.js

```diff
@@ -39,27 +41,54 @@ const initAccount = async (settings, i) => {
     config,
   })
 
-  accounts.push(
-    Account({
-      config,
-      randomInputSeed,
-      randomChangeSeed,
-      cryptoProvider,
-      blockchainExplorer,
-      accountIndex,
-    })
-  )
+  return Account({
+    config,
+    randomInputSeed,
+    randomChangeSeed,
+    cryptoProvider,
+    blockchainExplorer,
+    accountIndex,
+  })
 }
 
-before(async () => {
-  await Promise.all(accountSettings.map(initAccount))
+before(() => {
+  ADALITE_CONFIG.ADALITE_CARDANO_VERSION = 'shelley'
+  ADALITE_CONFIG.ADALITE_NETWORK = 'MAINNET'
+  const mockNet = mockNetwork(ADALITE_CONFIG)
+  mockNet.mockBulkAddressSummaryEndpoint()
+  mockNet.mockGetAccountInfo()
+  mockNet.mockGetStakePools()
+  mockNet.mockGetConversionRates()
+  mockNet.mockUtxoEndpoint()
+  mockNet.mockPoolMeta()
+  mockNet.mockGetAccountState()
+  mockNet.mockAccountDelegationHistory()
+  mockNet.mockAccountStakeRegistrationHistory()
+  mockNet.mockWithdrawalHistory()
+  mockNet.mockRewardHistory()
+  mockNet.mockPoolRecommendation()
+
+  Object.entries(accountSettings).forEach(([name, setting]) => {
+    accounts[name] = initAccount(setting)
```

Comment:
since initAccount() is async, this may not always work so should be still wrapped in Promise.all and await as I've written above

Reply:
Im not sure what you mean by 'this may not always work`. I decided to store Accounts as promises and they need to be awaited when the test is executing but in that case, everything should work always..right? :D

Reply:
ah, I see - what's the benefit of that though if you are awaiting on it anyway in tests below? Also seems like a chance to shoot ourselves in the foot in the future especially since the tests have not been migrated to ts yet so we can't add an explicit type annotation to the accounts object yet.

Reply:
You are right, I should have at least renamed it to something like `accountPromises`, anyway I ll refactor it to Promise.all and we don't need to await anything,